eth/beacon/deposit_helpers.py

```diff
@@ -0,0 +1,160 @@
+                current_slot
+        ):
+            return index
+    raise MinEmptyValidatorIndexNotFound()
+
+
+def validate_proof_of_possession(state: BeaconState,
+                                 pubkey: int,
+                                 proof_of_possession: bytes,
+                                 withdrawal_credentials: Hash32,
+                                 randao_commitment: Hash32) -> None:
+    deposit_input = DepositInput(
+        pubkey=pubkey,
+        withdrawal_credentials=withdrawal_credentials,
+        randao_commitment=randao_commitment,
+        proof_of_possession=EMPTY_SIGNATURE,
+    )
+
+    is_valid_signature = bls.verify(
+        pubkey=pubkey,
+        # TODO: change to hash_tree_root(deposit_input) when we have SSZ tree hashing
+        message=deposit_input.root,
+        signature=proof_of_possession,
+        domain=get_domain(
+            state.fork_data,
+            state.slot,
+            SignatureDomain.DOMAIN_DEPOSIT,
+        ),
+    )
+
+    if not is_valid_signature:
+        raise ValidationError(
+            "BLS signature verification error"
+        )
+
+
+def add_pending_validator(state: BeaconState,
+                          validator: ValidatorRecord,
+                          zero_balance_validator_ttl: int) -> Tuple[BeaconState, int]:
+    """
+    Add a validator to the existing minimum empty validator index or
+    append to ``validator_registry``.
+    """
+    # Check if there's empty validator index in `validator_registry`
+    try:
+        index = get_min_empty_validator_index(
+            state.validator_registry,
+            state.slot,
+            zero_balance_validator_ttl,
+        )
+    except MinEmptyValidatorIndexNotFound:
+        index = None
+
+    if index is None:
+        # Append to the validator_registry
+        validator_registry = state.validator_registry + (validator,)
+        state = state.copy(
+            validator_registry=validator_registry,
+        )
+        index = len(state.validator_registry) - 1
+    else:
+        # Use the empty validator index
+        state = state.update_validator(index, validator)
+
+    return state, index
+
+
+def process_deposit(*,
+                    state: BeaconState,
+                    pubkey: int,
+                    deposit: int,
+                    proof_of_possession: bytes,
+                    withdrawal_credentials: Hash32,
+                    randao_commitment: Hash32,
+                    zero_balance_validator_ttl: int) -> Tuple[BeaconState, int]:
+    """
+    Process a deposit from Ethereum 1.0.
+    """
+    validate_proof_of_possession(
+        state,
+        pubkey,
+        proof_of_possession,
+        withdrawal_credentials,
+        randao_commitment,
+    )
+
+    validator_pubkeys = tuple([v.pubkey for v in state.validator_registry])
+    if pubkey not in validator_pubkeys:
+        validator = ValidatorRecord.get_pending_validator(
```

Comment:
Would this be clearer as `create_pending_validator` rather than `get_`?
`get` implies to me that it is trying to retrieve.

Not sure what the codebase naming conventions are in general. Will defer to @pipermerriam